KHWEBA-3. Table-transactions

Vladyslav-Plakhuta opened the pull request:
![image](https://user-images.githubusercontent.com/34074565/37341050-ced4c0c2-26c8-11e8-8151-81fe02428b8d.png)

![image](https://user-images.githubusercontent.com/34074565/37341102-ee728fcc-26c8-11e8-8ee4-18dd58306876.png)

Vladyslav-Plakhuta committed:
upd

Vladyslav-Plakhuta committed:
upd

Vladyslav-Plakhuta committed:
changed for original

Vladyslav-Plakhuta committed:
append my component

Vladyslav-Plakhuta committed:
add link to scss file with styles

Vladyslav-Plakhuta committed:
add main buxy-table-transactions html class

Vladyslav-Plakhuta committed:
updated

Vladyslav-Plakhuta committed:
updated

Vladyslav-Plakhuta committed:
style's changes

gift-a commented on src/components/buxy-table-transactions/buxy-table-transactions.scss:
```diff
@@ -0,0 +1,57 @@
+    background-color: #f5f5f5;
+  }
+  .buxy-table-transactions-thead__tr:first-child {
+    background-color: #7575a3;
```
I think you should use Material Components Theme here to make this color:
add class "mdc-theme--primary-bg" for theme primary background and class "mdc-theme--text-primary-on-primary" for the text of your header.
https://material.io/components/web/catalog/theme/

why? because if we decide to make BUXY with green color, the whole website theme will become green excluding this table header.

Vladyslav-Plakhuta replied:
for first, I'm not sure to be justified to use framework just for correction of one element' bg inside component.
for second, I don't undestand about green color... just noticed that 'dialog for adding transaction' and 'Drawer' headers are close to violet

Vladyslav-Plakhuta replied:
as a variant:
 #751aff or #6600ff close to primary from MC

gift-a replied:
It is just a proposal to give this table the same behavior as other components have. To add styles like font family, font size, font color theme, background color theme, border colors via MDC classes instead of direct styles assigning. It's not just about bg color of header , it's about all of table components.

Vladyslav-Plakhuta committed:
add min-width for table for adaptive purpuses

Vladyslav-Plakhuta committed:
markup' fixing

Vladyslav-Plakhuta committed:
markup'fixing

Vladyslav-Plakhuta committed:
cell's content changes

OlgaOrlova commented on src/components/app/app.js:
```diff
@@ -1,11 +1,24 @@
 import template from "./app.html";
+import { buxyTableTransactionsComponent } from "../buxy-table-transactions/buxy-table-transactions";
```
please, name buxyTableTransactionsComponent from capital letter

ihor7016 replied:
++

ihor7016 commented on src/components/app/app.html:
```diff
@@ -1 +1,3 @@
-<div class="app">hi, <%= name %>!</div>
+<div class="app">
+  <div class="app-buxy-table-transactions"></div>
```
please use only BEM style class names

ihor7016 commented on src/components/buxy-table-transactions/buxy-table-transactions.html:
```diff
@@ -0,0 +1,55 @@
+
```
why empty spaces here?

ihor7016 commented on src/components/buxy-table-transactions/buxy-table-transactions.html:
```diff
@@ -0,0 +1,55 @@
+      <td class='buxy-table-transactions-tbody__td'>- 1,400.00</td>
+      <td class='buxy-table-transactions-tbody__td' colspan="4">Rent</td>
+      <td class='buxy-table-transactions-tbody__td'>Rent</td>
+      <td class='buxy-table-transactions-tbody__td'>
```
please use double quotes in html

ihor7016 commented on src/components/buxy-table-transactions/buxy-table-transactions.js:
```diff
@@ -0,0 +1,11 @@
+import template from "./buxy-table-transactions.html";
+
+export class buxyTableTransactionsComponent {
```
class names should start from Capital letter

ihor7016 replied:
please do not start component name from name of the application -- so it should be without 'buxy'

ihor7016 commented on src/components/buxy-table-transactions/buxy-table-transactions.scss:
```diff
@@ -0,0 +1,58 @@
+  box-sizing: border-box;
+}
+
+.buxy-table-transactions-thead {
```
it should be an element of the block

ihor7016 commented on src/components/buxy-table-transactions/buxy-table-transactions.scss:
```diff
@@ -0,0 +1,58 @@
+  font-family: $font-table-transactions;
+  font-size: $font-size-table-transactions;
+
+  .buxy-table-transactions-thead__th {
```
do not use cascades

ihor7016 commented on src/components/buxy-table-transactions/buxy-table-transactions.scss:
```diff
@@ -0,0 +1,58 @@
+  font-family: $font-table-transactions;
+  font-size: $font-size-table-transactions;
+
+  .buxy-table-transactions-tbody__td {
```
do not use cascades

ihor7016 commented on src/index.scss:
```diff
@@ -3,4 +3,5 @@
 
 @import "./components/page/page";
-@import "./components/app/app";
+//@import "./components/app/app";
```
do not comment files

ihor7016 reviewed (commented):
please make it look more material like  see https://getmdl.io/components/index.html#tables-section

Vladyslav-Plakhuta commented:
use table's style as in MDL, remove title of the project from name of the classes, replace double quotes in html, remove comments   
![screenshot](https://user-images.githubusercontent.com/34074565/37240989-3192e06a-245b-11e8-8a20-628161252fda.png)

Vladyslav-Plakhuta committed:
upd according to remarks

Vladyslav-Plakhuta committed:
upd according to remarks

Vladyslav-Plakhuta committed:
remove buxy from files' name

Vladyslav-Plakhuta committed:
rename

Vladyslav-Plakhuta committed:
rename

Vladyslav-Plakhuta committed:
rename

Vladyslav-Plakhuta commented:
review my task pls tried to fix all remarks

ihor7016 commented:
there should be a shadow around table

![image](https://user-images.githubusercontent.com/36405386/37242517-7aacb7ae-2473-11e8-8a97-3ee916483e8c.png)

and top row should not be dark

ihor7016 commented on src/components/variables.scss:
```diff
@@ -0,0 +1,2 @@
+$font-table-transactions: Roboto, Arial, sans-serif;
```
please make this variable name more common -- e.g. `$base-font-family`

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,47 @@
+
+
+
+
```
why empty lines here?

ihor7016 commented on src/components/table-transactions/table-transactions.html:
```diff
@@ -0,0 +1,43 @@
+      <td class="table-transactions__td">Rent</td>
+      <td class="table-transactions__td">
+        <span class="table-transactions__span">Amex Credit Card</span>
+        <div class="table-transactions__three-dots"></div>
```
three dots should be a button -- so you can click on it. You should also use three dots icon from material components

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,47 @@
+.table-transactions{
```
are your pre-commit hooks working? there should be a space between classname and `{`

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,47 @@
+  }
+
+  &__three-dots:after {
+    content: "\2807";
```
you should use three dots icon from material components

Vladyslav-Plakhuta committed:
added material icons link

Vladyslav-Plakhuta committed:
corrected code according to remarks

gift-a commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -1,12 +1,22 @@
+  box-shadow: 0px 10px 10px -10px #888,
+              10px 0px 10px -11px #888,
+              -10px 0px 10px -11px #888;
 
```
You can do this with MDC Elevation. just add a class mdc-elevation--z2 and import as you did with button
https://material-components-web.appspot.com/elevation.html
and here are principles of choosing shadow
https://material.io/guidelines/material-design/elevation-shadows.html#elevation-shadows-elevation-android

Vladyslav-Plakhuta commented:
drive three dots icon in button from MC, remove empty lines throughout .scss, replace variable's name for more common

review again please

gift-a reviewed (commented):
you should install your button with dependencies
`npm install --save @material/button`

it doesn't work yet.

Vladyslav-Plakhuta commented:
I've installed my button with dependencies 
npm install --save @material/button
checks another one time works good
![image](https://user-images.githubusercontent.com/34074565/37272780-6548b854-25e0-11e8-9aa4-2155b4defff2.png)
![image](https://user-images.githubusercontent.com/34074565/37272794-718d4d64-25e0-11e8-92c6-1a349555411b.png)

gift-a commented:
if you had done `npm install --save @material/button`, you would have had it in dependencies in you package.json. but you haven't.
![image](https://user-images.githubusercontent.com/14272396/37273640-1514cffa-25e3-11e8-8d01-e84f983b6c9d.png)
https://github.com/ihor7016/buxy-vanilla/blob/KHWEBA-3/package.json
maybe you've installed it before, when checked our branches. because npm_modules are ignored by git, they don't change after you change branch.

PS. to check it you can clone the repo in a new folder

gift-a commented:
looks like you will be merged next. add what's new in master to your branch and resolve conflicts. it will take a time and we will be able to review and approve your pull request with table in right place.

UPD you will not need to do `npm install --save @material/button` in this case))

ihor7016 commented:
![image](https://user-images.githubusercontent.com/36405386/37277061-45ca5210-25ec-11e8-9009-dc432f56ac90.png)
 
* please take a closer look at the table in MDL -- copy the shadow and font size from there
* look at the shape of 3dot buttons in a sidebar and make your button the same shape
* align your 3dot buttons to the right
* heading row should not be highlighted when hovering mouse

ihor7016 commented on src/components/page/page.scss:
```diff
@@ -1,3 +1,4 @@
 .page {
+  width:98%;
```
why are you setting page (body) width to 98%?

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,50 @@
+  box-sizing: border-box;
+  font-family: $base-font-family;
+  color: #8c8c8c;
+  -moz-box-shadow: 0px 10px 10px -10px #888,
```
you do not need these prefixed versions

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,50 @@
+.table-transactions {
+  border: 1px solid #ddd;
+  border-collapse: collapse;
+  width:100%;
```
please run formatter through your code (and make sure that you have precommit hooks working)

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,50 @@
+              -10px 0px 10px -11px #888;
+
+  &__tr {
+    height: 40px;
```
no sure that you need to set height of the row -- better use padding in cells

ihor7016 commented on src/components/table-transactions/table-transactions.scss:
```diff
@@ -0,0 +1,50 @@
+  min-width: 600px;
+  text-align: left;
+  box-sizing: border-box;
+  font-family: $base-font-family;
```
no sure that you need to set font family (we have got now enabled roboto by all app by default)

Vladyslav-Plakhuta committed:
added dependencies to json

Vladyslav-Plakhuta committed:
updated

Vladyslav-Plakhuta commented:
fixed next:
1. removed box-shadow property and added shadows from MC(mdc-elevation--z2) which similar to the sample from MDL
2. aligned 3dot buttons to the right
3. changed shape of 3dot buttons
4. abolished highlighting of heading row 
5. .page(body) width set to 100%
6. removed font-variable

review please

gift-a commented:
you have conflicts. you should resolve them. your table have to be in content container

Vladyslav-Plakhuta committed:
updated

sourmotion committed:
corrected table's width and button shape

sourmotion committed:
beautify

sourmotion committed:
resolve conflicts

Vladyslav-Plakhuta commented:
done all required changes
review please
